packages/eslint-plugin/tests/rules/ban-types.test.ts

```diff
@@ -388,6 +388,39 @@ let b: Foo<NS.Good>;
         },
       ],
     },
+
+    {
+      code: noFormat`
+declare let j: unknown;
+let m3 = { ...(j as Record<string, never>) };
+      `,
+      output: `
+declare let j: unknown;
+let m3 = { ...(j as object) };
+      `,
+      options: [
+        {
+          types: {
+            'Record<string, never>': {
+              message: 'Use object instead.',
+              fixWith: 'object',
+            },
+          },
+        },
+      ],
+      errors: [
+        {
+          messageId: 'bannedTypeMessage',
+          data: {
+            name: 'Record<string,never>',
+            customMessage: ' Use object instead.',
+          },
+          line: 3,
+          column: 21,
+        },
+      ],
+    },
+
```

Comment:
There's no need for this test. It isn't testing anything new - this file already tests that providing `types` options are respected.

```suggestion

```